### DISCUSSION TITLE
Failed to processing OIDC by oidc provider via Authelia

### DISCUSSION THREAD

**Discussion by catfishlty** [pocketbase/pocketbase#3732] [Q&A] [answered]
Hi~
I used PocketBase as golang package to start a app.
I've set OIDC configs in admin UI, and add OIDC key info in [Authelia](https://www.authelia.com/integration/openid-connect/introduction/)

I add OAuth Login logic in web, it can open external window and show Authelia Login Page.
But after I login successfully, it will return 400 error.

I've start a debug mode to trace the data, and I found that `authorization` step is success, and the redirect is also working well, but the `Token` step has some error with 401 response from Authelia.
## Error Response from Authelia Token endpoint.
```
{"error":"invalid_client","error_description":"Client authentication failed (e.g., unknown client, no client authentication included, or unsupported authentication method)."}
```
## Error Logs From Authelia
```
time="2023-11-10T16:59:20+08:00" level=error msg="Access Request failed with error: Client authentication failed (e.g., unknown client, no client authentication included, or unsupported authentication method). the passwords don't match" method=POST path=/api/oidc/token remote_ip={ip} stack="github.com/authelia/authelia/v4/internal/handlers/handler_oidc_token.go:27                   OpenIDConnectTokenPOST\ngithub.com/authelia/authelia/v4/internal/middlewares/http_to_authelia_handler_adaptor.go:113 NewHTTPToAutheliaHandlerAdaptor.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/bridge.go:54                            (*BridgeBuilder).Build.func1.1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:35                           SecurityHeadersNoStore.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:25                           SecurityHeadersCSPNone.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:16                           SecurityHeaders.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/cors.go:216                             (*CORSPolicy).Middleware.func1\ngithub.com/fasthttp/router@v1.4.14/router.go:414                                             (*Router).Handler\ngithub.com/valyala/fasthttp@v1.43.0/http.go:154                                              (*Response).StatusCode\ngithub.com/valyala/fasthttp@v1.43.0/server.go:2338                                           (*Server).serveConn\ngithub.com/valyala/fasthttp@v1.43.0/workerpool.go:224                                        (*workerPool).workerFunc\ngithub.com/valyala/fasthttp@v1.43.0/workerpool.go:196                                        (*workerPool).getCh.func1\nruntime/asm_amd64.s:1594
```

I'm very sure the settings are all correct, so plz help check this ,Thanks~

**Comment by ganigeorgiev** [comment]
Without a code sample how you are sending the request and what flow do you use I'm not sure how to help.

Based on the error message I guess that invalid provider name was used. In your case since you use an OIDC OAuth2 app, the provider value must be `oidc`.

If that doesn't help, please provide more details about your setup and ideally some code sample showing what you are doing.

**Comment by catfishlty** [comment, marked as answer]
@ganigeorgiev 
Sorry for late reply~
I confirmed that the issue was caused by a wrong `client_secret`, maybe a typo or something.

Thank you for your help~

**Comment by catfishlty** [comment]
Follow https://www.authelia.com/integration/openid-connect/pocketbase/ to integrate with Authelia `OIDC` 🚀
